liquidprompt

```diff
@@ -1399,6 +1399,8 @@ _lp_color_map() {
     scale=${2:-100}
     # Transform the value to a 0..${#COLOR_MAP} scale
     idx=_LP_FIRST_INDEX+100*$1/scale/${#LP_COLORMAP[*]}
+    # When off the scale, use the "highest" available color
+    (( idx < ${#LP_COLORMAP[*]} )) || idx=-1
```

Comment:
It's actually slightly less of a winner; this negative index notation was added to Bash in version 4.2:
http://mywiki.wooledge.org/BashFAQ/061. Which means we can't use it.

Don't worry about fixing it, I'm working on a fix for the color scale right now for 1.12 that will include this.

Reply:
Well, the fix for that is really trivial, so I pushed it nonetheless. `idx=${#LP_COLORMAP[*]}-1` instead of just `idx=-1`.

That should be as backwards compatible as the rest of liquidprompt already is, considering that the `${#name[*]}` syntax to get the number of elements in an array is already used elsewhere.